src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -172,47 +203,30 @@ private void handle(String consumeTag, Delivery delivery) {
         }
     }
 
-    private boolean resubscribe() {
-        LOGGER.info("Trying to resubscribe to queue.");
-
-        RabbitMQConfiguration configuration = this.connectionManager.get().getConfiguration();
-        int attemptsCount = 0;
-        int amountOfAttempts = configuration.getMaxRecoveryAttempts();
-        boolean successful = false;
-
-        while (attemptsCount < amountOfAttempts) {
-            try {
-                Thread.sleep(configuration.getMaxConnectionRecoveryTimeout());
-            } catch (InterruptedException | IllegalArgumentException e) {
-                LOGGER.error(e.getMessage());
-            }
-
-            try {
-                attemptsCount++;
-                consumerMonitor.set(null);
-                start();
-
-                CommonMetrics.setReadiness(true);
-                successful = true;
-                break;
-            } catch (Exception e) {
-                if(attemptsCount >= amountOfAttempts) {
-                    LOGGER.error("Failed to start listening when resubscribing after {} attempts.", amountOfAttempts);
+    private void recursiveResubscribe() {
+        try {
+            if(futureStart().get()) {
```

Comment:
Call of the get method on a future blocks current thread. 
Please move logic to updated probe monitors to the task related to recovery subscription